### PR TITLE
(WIP) Support soft delete using gorm and various test cases.

### PR DESCRIPTION
**Pull request by pranavgore09**
Added WIP because this should be merged after #2. Otherwise it is complete.

`This PR has schema change.`

WHAT:
Implemented soft delete using gorm.
Added tests for the same. Integration test added in order to verify that user record is created, fetched, returned as expected.
Added tests for UserFilterByIdentity and related methods.

WHY:
Allow user to create a user record with same email if previous record is deleted.
Record is considered in the table when deleted_at column has NULL value.

HOW:
Created `combination key index` on `email, deleted_at`.
This is not sufficient because postgresql treats every NULL as a different value and NULL in record1 != NULL in record2.
Hence added logic support while creating the item. It will check (active)user's existance

### CODE REVIEW

File: account/user.go [hunk 1]
```diff
@@ -67,19 +68,36 @@ func (m *GormUserRepository) Load(ctx context.Context, id uuid.UUID) (*User, err
...
+		err := m.db.Create(u).Error
+		if err != nil {
+			goa.LogError(ctx, "error adding User", "error", err.Error())
+			return err
+		}
 
-	return nil
+		return nil
+	}
+	return errors.New("Record already exist")
```
Comment (baijum) [inline account/user.go:100]: It is better to use error object implemented the `error` interface: https://golang.org/pkg/builtin/#error

File: token/token.go [hunk 2]
```diff
@@ -56,6 +56,11 @@ func (mgm tokenManager) Extract(tokenString string) (*account.Identity, error) {
 		return nil, errors.New("Token not valid")
 	}
 
+	claimedUUID := token.Claims.(jwt.MapClaims)["uuid"]
+	if claimedUUID == nil {
+		return nil, errors.New("UUID can not be nil")
```
Comment (baijum) [inline token/token.go:61]: It is better to use error object implemented the `error` interface: https://golang.org/pkg/builtin/#error

File: account/user_test.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	resource.Require(t, resource.Database)
+	// this test makes sure that UserByEmails eliminates deleted entries
+	ctx := context.Background()
+	userRepo := account.NewUserRepository(db)
+	identityRepo := account.NewIdentityRepository(db)
+	identity := account.Identity{
+		FullName: "Test User Integration 123",
+		ImageURL: "http://images.com/42",
+	}
+	email := "primary@test.com"
```
Comment (baijum) [inline account/user_test.go:55]: Let's not spam that domain - it's a real domain. Use `primary@example.com`.

**Comment by aslakknutsen** [comment]
Merged to https://github.com/aslakknutsen/almighty-core/tree/user_auth

**Pull request closed without merging**